### PR TITLE
FP-007-getAll-byID

### PR DESCRIPTION
**Pull request by TesaRocks**
Diego, please have a look at the new features:
Get all Products (I originally pushed it to main by mistake)
IProducts Interface(I originally pushed it to main by mistake)
Products Routing (I originally pushed it to main by mistake) 
and get productsby ID (shown in this branch)
Please let me know.
PS. There s hard code 1 number in one case only Will modify later.

### CODE REVIEW

File: src/repositories/user.repository.ts [hunk 1]
```diff
@@ -50,4 +50,26 @@ export class UserRepository {
     });
     return productList;
   }
+  public async getProductById(index: number): Promise<IProduct> {
+    const productFound = await this.db.query({
+      sql: `SELECT * FROM products WHERE productId= ${index}`,
```
Comment (diegolaciar) [inline src/repositories/user.repository.ts]: Check if it is possible to use a params property on this function, check what brings the driver from the driver.
This kind of instruction is Highly insecure as could lead to an SQL Injection!

File: src/repositories/user.repository.ts [hunk 2]
```diff
@@ -50,4 +50,26 @@ export class UserRepository {
...
+    });
+    return productFound[0];
+  }
+  public async saveProduct(product: IProduct, id: number): Promise<IProduct> {
+    const okPacket: OkPacket = await this.db.query({
+      sql: `INSERT INTO products (name, description, imagePath, price, id) VALUES('${product.name}', '${product.description}', '${product.imagePath}', '${product.price}', '${id}');`,
```
Comment (diegolaciar) [inline src/repositories/user.repository.ts]: SQL injection here as well, looks very bad to use the driver this way.

File: src/services/userV2.service.ts [hunk 3]
```diff
@@ -26,4 +26,13 @@ export class UserV2Service {
   public async getAllProducts() {
     return await this.userRepository.getProducts();
   }
+  public async getProductById(id: number) {
```
Comment (diegolaciar) [inline src/services/userV2.service.ts]: Not a go here mate.

You can put products on a USers file.
Just make a new file
products.service.ts
and work everything there.

File: src/repositories/user.repository.ts [hunk 4]
```diff
@@ -50,4 +50,26 @@ export class UserRepository {
     });
```
Comment (diegolaciar) [inline src/repositories/user.repository.ts]: user.repository.ts
you MUST code more consistent on the design of your artifacts, your sources.
make a new file. products.repositoy.ts in this case. Put your product code there.

**Review by diegolaciar — CHANGES_REQUESTED** [review]
Check the comments mate.

Cheers,
Diego

File: src/repositories/user.repository.ts (hunk 1 above)
Comment (TesaRocks) [reply src/repositories/user.repository.ts]: Hi ! not very sure what you mean, I change the * to 
SELECT productId,name, description, imagePath, price, created, sold, id FROM products
Is this what you mean?
In the users feature, we implemented it that way.

File: src/repositories/user.repository.ts (hunk 2 above)
Comment (TesaRocks) [reply src/repositories/user.repository.ts]: This is the same exact way as we did it in the user feature. Can you please give me a hand?

File: src/services/userV2.service.ts (hunk 3 above)
Comment (TesaRocks) [reply src/services/userV2.service.ts]: Done, modified, thanks

File: src/repositories/user.repository.ts (hunk 4 above)
Comment (TesaRocks) [reply src/repositories/user.repository.ts]: This was my initial concern I wrote you on the first hand.
Does it make sense to have 2 connections in parallel to the DATABASE?Thanks

File: src/repositories/user.repository.ts (hunk 1 above)
Comment (diegolaciar) [reply src/repositories/user.repository.ts]: I'm not sure how to do it.  is for research.  but certainly, this is a serious fault to security.  this kind of instruction on code is a target for a SQL Injection attack.

Comment (diegolaciar) [reply src/repositories/user.repository.ts]: I"m talking about the way you interpolate strings. without any kind of filter. for the string that comes from the client.

Comment (diegolaciar) [reply src/repositories/user.repository.ts]: that allows anyone to inject, I mean send prepared kind of injection, vectors into your database.

File: src/repositories/user.repository.ts (hunk 4 above)
Comment (diegolaciar) [reply src/repositories/user.repository.ts]: No it doesn't  have sense,  you must inject the same connection to each repository

File: src/repositories/user.repository.ts (hunk 1 above)
Comment (TesaRocks) [reply src/repositories/user.repository.ts]: Done, please have a look. Thanks

File: src/routers/product.router.ts [hunk 5]
```diff
@@ -29,7 +29,7 @@ productRouter.get("/:id", async (req: Request, res: Response) => {
 });
 
 productRouter.post("", async (req: Request, res: Response) => {
-  const userId: number = 1;
+  const userId: number = 5;
```
Comment (diegolaciar) [inline src/routers/product.router.ts]: I don't get why this line ?

**Review by diegolaciar — APPROVED** [review]
Approved, but I left a comment there..
check it out.

File: src/repositories/user.repository.ts (hunk 2 above)
Comment (TesaRocks) [reply src/repositories/user.repository.ts]: Its a temporary solution until the architecture of the front end is done. thnaks

File: src/routers/product.router.ts (hunk 5 above)
Comment (TesaRocks) [reply src/routers/product.router.ts]: Its a temporary solution until the architecture of the front end is done. thnaks

**Review by diegolaciar — APPROVED** [review]

**Comment by diegolaciar** [comment]
Ship it !

**Pull request merged**